### PR TITLE
win_scheduled_task: rewrite

### PR DESCRIPTION
**Pull request by jborean93**
##### SUMMARY
This is a WIP branch that contains the changes I've made so far to win_scheduled_task. Some of the things I have changed are

* Now using the COM object Schedule.Service so this will work with Server 2008/R2 and Windows 7
* Added support for all trigger types including logon, boot, registration change and so on
* Modified the actions and triggers so that you can specify multiple ones instead of just 1
* More tests than before to cover more of the code
* Idempotency for task settings where before it would delete and recreate the task even when it wasn't needed
* Various fixes for bugs listed below this
* Diff support to show the changes in the task

Issues this should ultimately solve:
* https://github.com/ansible/ansible/issues/28039 - just a general bug fix
* https://github.com/ansible/ansible/issues/27248 - adds the boot trigger
* https://github.com/ansible/ansible/issues/25478 - disable an existing task without specifying all the metadata around it
* https://github.com/ansible/ansible/issues/23649 - more fine grained control over triggers now enables
* https://github.com/ansible/ansible/issues/23472 - idempotency fixes

What I am hoping to get a consensus from everyone with this PR being up right now is
* The module is now massive, should it be split up into different modules to handle things like triggers
* I've tried to make sure it is as close to backwards compatible to the current devel version but there is bound to be things I've missed or will come up when i do the testing cycle. Do we care, should this be a separate module and we deprecated `win_scheduled_task` later on?
* The way the original module handled triggers and actions was quite organic, this is pretty incompatible when it comes to specifying multiple triggers and actions which I've tried to handle
* There are lots of options, especially with settings, should this be split out or have a single `settings` option which takes in a dict of settings, similar to how IIS is set up.

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
win_scheduled_task

##### ANSIBLE VERSION
```
2.4
```

cc @dagwieers @jhawkesworth @nitzmahone @andrewsaraceni for input on this.

~~Note: This also requires https://github.com/ansible/ansible/issues/30480 to be fixed so it can handle empty and single action/trigger lists.~~ done

### CODE REVIEW

**Comment by jborean93** [comment]
Ok, the rewrite is complete and tests pass with the changes. If anyone has some free time it would be great if you could review it and try it out.

File: lib/ansible/modules/windows/win_scheduled_task.ps1 [hunk 1]
```diff
@@ -1,254 +1,1146 @@
...
 
+# try and get the task at the path
+$task = Test-TaskExists -task_folder $task_folder -name $name
+
+$task_path = $null
+if ($path.EndsWith("`\")) {
+    $task_path = "$path$name"
+} else {
+    $task_path = "$path\$name"
+}
```
Comment (dagwieers) [inline lib/ansible/modules/windows/win_scheduled_task.ps1]: Isn't there something similar to `os.path.join()` that would abstract this away in PS ?

**Review by dagwieers — APPROVED** [review]

File: lib/ansible/modules/windows/win_scheduled_task.ps1 (hunk 1 above)
Comment (jborean93) [reply lib/ansible/modules/windows/win_scheduled_task.ps1]: Looks like `Join-Path` does the trick, I'll update the PR.

Thanks for the suggestion.

**Comment by jborean93** [comment]
consensus from the WWG today was to merge this as is so we can get some wider testing from devel users.

**Pull request merged**